app/src/main/java/com/github/orkest/View/sharing/PlaylistActivity.kt

```diff
@@ -0,0 +1,153 @@
+package com.github.orkest.View.sharing
+
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.foundation.lazy.items
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.content.ContextCompat.startActivity
+import com.github.orkest.Model.DeezerApiIntegration
+import com.github.orkest.Model.Song
+import com.github.orkest.View.sharing.ui.theme.OrkestTheme
+import com.github.orkest.ViewModel.playlist.PlaylistViewModel
+import com.spotify.android.appremote.api.ConnectionParams
+import com.spotify.android.appremote.api.Connector
+import com.spotify.android.appremote.api.SpotifyAppRemote
+
+/*
+    * This class is used to display the list of songs that the user share with other users.
+ */
+class PlaylistActivity() : ComponentActivity() {
+
+//
+
+    private val playlistViewModel = PlaylistViewModel()
+    private var spotifySongId = ""
+    private var mSpotifyAppRemote: SpotifyAppRemote? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // -------Connect to Spotify AppRemote-------
+        // TODO include spotify in addition to Deezer
+
+        //--------Update Database---------
+
+        // check if extras are not null
+        Log.d("PlaylistActivity", "songId: ${intent.getStringExtra("songId")}")
+        intent.getStringExtra("songName")?.let {
+            // update database
+            val song = Song(
+                Title = it,
+                Artist = "Unknown Artist",
+                Album = "Unknown Album",
+                URL = "Unknown ID")
+
+            // store song in a way that both users can see them in their pages
```

Comment:
Isn't this page used to display received songs? Why would there be storing on startup?

Reply:
Ah never mind, I understood by reading the rest of the PR, maybe a comment would be nice to explicit that this is for the case when we share music from the provider into orkest, or put it in a separate function to make it more readable